.github/workflows/app-publish.yaml

```diff
@@ -0,0 +1,121 @@
+    branches:
+      - master
+      - release*
+
+jobs:
+  build:
+    runs-on: ubuntu-20.04
+    if: ${{ startsWith(github.ref, 'refs/tags/v') || github.repository == 'jellyfin/jellyfin-android' }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+
+      - name: Setup Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Set JELLYFIN_VERSION
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: echo "JELLYFIN_VERSION=$(echo ${GITHUB_REF#refs/tags/v} | tr / -)" >> $GITHUB_ENV
+
+      - name: Build Release
+        run: ./gradlew --no-daemon --build-cache assembleRelease versionTxt
+
+      - name: Split APK release types
+        run: |
+          mkdir app/build/outputs/apk/libre_release app/build/outputs/apk/proprietary_release;
+          mv app/build/outputs/apk/**/jellyfin-android-*-libre-release.apk app/build/outputs/apk/libre_release/;
+          mv app/build/outputs/apk/**/jellyfin-android-*-proprietary-release.apk app/build/outputs/apk/proprietary_release/;
+
+      - uses: actions/upload-artifact@v2
+        with:
+          name: build-publish-artifact
+          retention-days: 30
+          if-no-files-found: error
+          path: |
+            app/build/outputs/apk/libre_release/*
+            app/build/outputs/apk/proprietary_release/*
+            app/build/version.txt
+
+  publish-release:
+    runs-on: ubuntu-20.04
+    needs: build-publish
+    if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+    steps:
+      - uses: actions/download-artifact@v2
+        with:
+          name: build-publish-artifact
+
+      - name: Set JELLYFIN_VERSION
+        if: ${{ startsWith(github.ref, 'refs/tags/v') }}
+        run: echo "JELLYFIN_VERSION=$(echo ${GITHUB_REF#refs/tags/v} | tr / -)" >> $GITHUB_ENV
+
+      - name: Sign Libre Release APK
+        uses: r0adkll/sign-android-release@v1
+        id: sign_libre
+        with:
+          releaseDirectory: app/build/outputs/apk/libre_release
+          signingKeyBase64: ${{ secrets.SIGNING_KEY }}
+          alias: ${{ secrets.ALIAS }}
+          keyStorePassword: ${{ secrets.KEY_STORE_PASSWORD }}
+          keyPassword: ${{ secrets.KEY_PASSWORD }}
+
+      - name: Sign Proprietary Release APK
+        uses: r0adkll/sign-android-release@v1
+        id: sign_proprietary
+        with:
+          releaseDirectory: app/build/outputs/apk/proprietary_release
+          signingKeyBase64: ${{ secrets.SIGNING_KEY }}
+          alias: ${{ secrets.ALIAS }}
+          keyStorePassword: ${{ secrets.KEY_STORE_PASSWORD }}
+          keyPassword: ${{ secrets.KEY_PASSWORD }}
```

Comment:
We might want to move the signing part to Gradle, We'd need to apply the `signingConfigs` in the `android {}` block for that though.

Reply:
:thinking: I understand what you mean, however have to disclose I would have no idea on how to do that. My experience with Gradle so far is rather limited, but I will see what I can do :+1:

Reply:
No problem, we can always change this part later

Reply:
Okay I read up on this a bit and have a couple of questions:

* given the logic of this, I assume the jks for the gradle based solution still needs to be base64-ed and serialized back to a jks before the build?
* given what I have found the signing would happen as part of the release build, if the jks, its passwd and the alias where present and unsigned if they are missing. However, I assume you thought of a custom gradle signing task that does nothing but sign the output?

Reply:
That's correct, you'd deserialize the keystore, put it to the right location, set the alias and passwords from the secrets, and the signing is done by the build tools. I did exactly that a while ago, see [here](https://github.com/Maxr1998/MaxLock/blob/master/buildSrc/src/main/kotlin/SigningConfigHelper.kt#L24) for reference.

Reply:
thank you for that reference, I shall give it a try to integrate it here.

Reply:
Okay, 412a4d58414d5f70f58206b9bcd679f1964a0416 now has a very 'scripted' variation of the sign config contained in the build.gradle.kts. However, I guess you imagined something more elegant than that, so please let me know.

Reply:
Okay, ended up installing Android Studio because IDEA refused to import properly and added a very 'scripted' signing config in a95b2e4c75adf799615286cfec5941dbacb563e4. I assume you where thinking of a more elegant solution, so let me know if I should change something or just remove the commit an squash the rest of the workflow commits.